client/src/App.js

```diff
@@ -38,29 +38,45 @@ function App() {
     return library.find((book) => book.id === focusedBook.id);
   }
 
-  function renderBookDetails(book) {
-    return (
-      <BookDetails
-        book={book}
-        onRemoveDetailView={() => setView('')}
-        onAddRating={addRating}
-      />
-    );
-  }
-
   function renderBookDetailsHelper(book) {
     setDetailedBook(book);
     setView('details');
   }
 
   function updateBook(property, value, bookToUpdate) {
-    const upDatedBooks = library.map((book) => {
+    const updatedBooks = library.map((book) => {
       if (book.id === bookToUpdate.id) {
         book[property] = value;
       }
       return book;
     });
-    setLibrary(upDatedBooks);
+    setLibrary(updatedBooks);
+  }
+
+  function updateBooksInCompartment(property, selection, book) {
+    const updatedShelves = shelves.map((shelf) => {
+      if (shelf.id === selection.bookshelf) {
+        shelf.columns.map((column) => {
+          if (column.id === selection.column) {
+            column.compartments.map((compartment) => {
+              if (compartment.id === selection.compartment) {
+                let existingBooks;
+                compartment[property]
+                  ? (existingBooks = compartment[property])
+                  : (existingBooks = '');
+                existingBooks === ''
```

Comment:
existingBooks is an array, right? 
```suggestion
                  : (existingBooks = []);
                existingBooks.length === 0
```